crates/sui-types/src/crypto.rs

```diff
@@ -696,14 +743,41 @@ impl<'de> Deserialize<'de> for Signature {
 
 // Can refactor this with a library
 impl Signature {
-    pub fn new<T>(value: &T, secret: &dyn signature::Signer<Signature>) -> Signature
+    pub fn new_legacy<T>(value: &T, secret: &dyn signature::Signer<Signature>) -> Signature
     where
         T: Signable<Vec<u8>>,
     {
         let mut message = Vec::new();
         value.write(&mut message);
         secret.sign(&message)
     }
+
+    /// Testing only
```

Comment:
Do we need to add a feature for this or a flag that it's only for testing? cc: @bmwill

Reply:
yeah thats the eventual goal. i can remove this for now, since I did not touch any existing test cases to use Intent. this PR aims to just introduce the `new_secure` and `verify_secure` for Signature and Authority without actually switching to them.